controllers/release/adapter.go

```diff
@@ -765,11 +765,79 @@ func (a *adapter) EnsureCollectorsProcessingResourcesAreCleanedUp() (controller.
 // Processing step are cleaned up once processing is finished. This exists in conjunction with EnsureFinalizersAreCalled because
 // the finalizers should be removed from the pipelineRuns even if the Release is not marked for deletion for quota reasons.
 func (a *adapter) EnsureReleaseProcessingResourcesAreCleanedUp() (controller.OperationResult, error) {
-	if !a.release.HasTenantPipelineProcessingFinished() || !a.release.HasManagedPipelineProcessingFinished() || !a.release.HasFinalPipelineProcessingFinished() {
-		return controller.ContinueProcessing()
+	var cleanupErrors []error
```

Comment:
This code could be optimized probably. If tenant processing has not finished, there's no need to check for the rest as they are executed in serial. Either split in different operations with proper gates or control the workflow here.

Reply:
Absolutely! Excellent catch, I missed it. Modified now.

Reply:
Excellent suggestion, it made the code a lot shorter. Thanks! 
done